src/Model/Anon.py

```diff
@@ -14,181 +16,114 @@
-        # print("Patient Sex - ", patient_sex)
-        # MD 5 hashing
-        hash_Study_Date_MD5 = uuid.uuid5(uuid.NAMESPACE_URL, Study_Date)
-        # Hashing the MD5 hash again using SHA1
-        hash_Study_Date_sha1 = uuid.uuid3(uuid.NAMESPACE_URL, str(hash_Study_Date_MD5))
-        # storing the hash to dataset
-        ds_rtss.StudyDate = str(hash_Study_Date_sha1)
-    else:
-        print("Patient Study_Date not found")
-
-    # -----------------content date date-----------
-    if "ContentDate" in ds_rtss:
-        Content_Date = str(ds_rtss.ContentDate)
-        # print("Patient Sex - ", patient_sex)
-        # MD 5 hashing
-        hash_Content_Date_MD5 = uuid.uuid5(uuid.NAMESPACE_URL, Content_Date)
-        # Hashing the MD5 hash again using SHA1
-        hash_Content_Date_sha1 = uuid.uuid3(
-            uuid.NAMESPACE_URL, str(hash_Content_Date_MD5)
-        )
-        # storing the hash to dataset
-        ds_rtss.ContentDate = str(hash_Content_Date_sha1)
-    else:
-        print("Patient Content_Date not found")
-
-    # ------------------Structure set date-----------------
-
-    if "StructureSetDate" in ds_rtss:
-        Structure_Set_Date = str(ds_rtss.StructureSetDate)
-        # print("Patient Sex - ", patient_sex)
-        # MD 5 hashing
-        hash_Structure_Set_Date_MD5 = uuid.uuid5(uuid.NAMESPACE_URL, Structure_Set_Date)
-        # Hashing the MD5 hash again using SHA1
-        hash_Structure_Set_Date_sha1 = uuid.uuid3(
-            uuid.NAMESPACE_URL, str(hash_Structure_Set_Date_MD5)
-        )
-        # storing the hash to dataset
-        ds_rtss.StructureSetDate = str(hash_Structure_Set_Date_sha1)
-    else:
-        print("Patient Content_Date not found")
-
-    # used to return flag = 1 to indicate the first file is used for saving the hash in
-    # hash_CSV file so CSV function will not be performed for rest of the files.
-    if file_no == 1:
-        if "PatientID" in ds_rtss:
-            P_name_ID = patient_name + " + " + patient_ID
-            print("Pname and ID=   ", P_name_ID)
-        else:
-            P_name_ID = patient_name + " + " + "PID_empty"
-            print("Pname and ID=   ", P_name_ID)
-        return (P_name_ID, hash_patient_name_sha1, 1)
-    else:
-        return (0, hash_patient_name_sha1, 0)
 
+def _hash_identifiers_in_place(ds_rtss):
+    """in place anonymisation of a set of identifiers in a dataset
+    Parameters
+    ----------
+    ds_rtss: ``pydicom.dataset.Dataset``
+        A dataset to be anonymised
 
-## ===================================CHECK FILE EXIST================================================
+    Returns
+    -------
+    """
+    mandatory_identifying_keys = [
+        "PatientName",
+        "PatientID",
+    ]
+
+    identifying_keys = [
+        "PatientName",
+        "PatientID",
+        "PatientBirthDate",
+        "PatientSex",
+        "InstanceCreationDate",
+        "StudyDate",
+        "ContentDate",
+        "StructureSetDate",
```

Comment:
only an RT SS will have Structure Set Date
So while it does need to be anonymised, if present, it is completely unremarkable if it is not present in CT data